### DISCUSSION TITLE
List/Search rule on "users" only if logged in as "server_users"

### DISCUSSION THREAD

**Discussion by Wavegreen** [pocketbase/pocketbase#4147] [Q&A] [answered]
I have a sveltekit server-side endpoint that checks a user before sending a paymentintent to stripe.
For this, I have created a new auth collection "server_side_users" to keep them separate from my normal "users" collection.
What rules/filters can I put into List/Search in "users" to only allow searching if a "server_side_users" auth is making the request?

I tried using the following list/search rule within "users":

@request.auth.id = <serversideuserID> 

but I think that it does not know that ID since it's in another collection?

I also had no luck with @request.auth.collectionId/collectionName.

Thanks in advance!

**Comment by ganigeorgiev** [comment]
> I also had no luck with @request.auth.collectionId/collectionName.

`@request.auth.collectionName = "server_side_users"` should work.

If it doesn't work, then make sure to check that your client is properly authenticated and that there is no cache involved.

If that doesn't help, please provide a code sample how you are sending the request and/or create a **minimal** reproducible repo.

**Reply by Wavegreen** [reply]
Thanks for your quick reply!
After some testing with the simple @request.auth.id != ""  I found that the server-side code which I thought was authenticating was in fact not doing that. I had placed the code block that was logging in, into every request. Thinking about it now, that was very silly since it is not very efficient to create a new RecordModel every time. Now the authwithpassword just happens once when starting. This caused the check in List/Search to work using @request.auth.collectionName = "server_side_users".
Thinking further, is this a problem? I am thinking about the login expiring if I don't restart my entire sveltekit app. Would it be advisable to somehow destroy the RecordModel and make a new one every 100th request or refresh it?

**Reply by ganigeorgiev** [reply, marked as answer]
I'm actually confused what is the purpose of `server_side_users`, how they are created and how they authenticate.

If you just need an admin level access to the collection, then you don't need an API rule (or even a new auth collection) for that. You can "lock" your collection action and interact with the PocketBase APIs as admin. For an example client, you could check https://github.com/pocketbase/pocketbase/discussions/3792#discussioncomment-7621130.

**Reply by Wavegreen** [reply]
I thought that it would be better to separate server-side accounts from admins who log into the UI. I changed everything over to admin and can now lock the permission. Thanks for your help!
